cypress/plugins/autorecord/index.ts

```diff
@@ -81,7 +81,22 @@ export function autoRecord() {
         const { url, method, body } = req
         const { host } = parseUrl(url, true)
         const status = res.statusCode
-        const data = res.body.constructor.name === 'Blob' ? blobToPlain(res.body) : res.body
+
+        let data = ''
+        switch (res.body.constructor.name) {
+          case 'Blob': {
+            data = blobToPlain(res.body)
+            break
+          }
+          case 'Object':
+          case 'Array': {
+            data = JSON.stringify(res.body)
+            break
+          }
+          default: {
+            data = res.body
+          }
+        }
```

Comment:
We can keep this pure by making it an IIFE, `const data = (() => {...`

Reply:
Good point